### PR TITLE
Update support of different checksums in fty-core scripts

### PR DESCRIPTION
**Pull request by jimklimov**
As elsewhere, these changes extend original support for accompanying `\*.md5` files with addition of `\*.sha256` and `\*.cksum` files (the latter far from secure, but adding easy to verify content length check, unlike the first two... and yet another algorithm to fool simultaneously in case of malicious intrusion).

### CODE REVIEW

**Comment by jimklimov** [comment]
Note that the task scope is not yet all complete at this point:

[ ] update-rc3 should require (option/default) that at least one checksum file is available (and all those available do match)
[ ] maybe same for ci-reset-virtual-machine.sh - but it is a developer aid, so not as strict
[ ] actually do check multiple checksum files at once in update-rc3
[ ] use the file length field from cksum files in update-rc3

File: tests/CI/ci-reset-virtual-machine.sh [hunk 1]
```diff
@@ -164,26 +164,133 @@ check_md5sum() {
...
+	case "$FILE_CKSUM" in
+		/*) ;;
+		*) FILE_CKSUM="`pwd`/$FILE_CKSUM" ;;
+	esac
+	if [ -s "$FILE_DATA" -a -s "$FILE_CKSUM" ]; then
+		logmsg_info "Validating OS image file '$FILE_DATA' against its CRC/cksum checksum '$FILE_CKSUM'..."
+		LENEXP="`awk '{print $2}' < "$FILE_CKSUM"`"
+		LENACT="`ls -lad "$FILE_DATA" | awk '{print $5}'`"
+		if [ x"$LENEXP" != x"$LENACT" ]; then
+			logmsg_error "Checksum (CRC/cksum) validation of '$FILE_DATA' against '$FILE_CKSUM' FAILED the lenght test!"
```
Comment (boricj) [inline tests/CI/ci-reset-virtual-machine.sh]: failed the leng**th** test

**Comment by jimklimov** [comment]
> 
> 
> I'm concerned about skipping validation when a checksum is skipped because it is missing. This seems like a fairly obvious way to pass an unauthenticated image through by nuking all checksums. Any reason why it is that way?

That is the "update-rc3 should require (option/default) that at least one checksum file is available" part ;)

So far the checksums were an aid against bad data transfers - we've had a fair share of short downloads, bad USB disks (and/or unplugged too early) etc., not an anti-deliberate-intrusion mechanism.

File: tests/CI/ci-reset-virtual-machine.sh (hunk 1 above)
Comment (jimklimov) [reply tests/CI/ci-reset-virtual-machine.sh]: Typo fixed

**Comment by jimklimov** [comment]
As of the latest commits, the code should download and check all 3 types of checksums; with a bit of a hack it also checks the file-length while verifying "cksum" ones.

The updated code still does not "honestly" handle file lengths otherwise (there is a field in checksum cache structure and code for that, but unused for now).

Also logged an idea to use the OS image manifest files where available as carriers of both a SHA256 checksum value and OS image file size, so with appropriate algo handling they can be seen as checksum-files for this purpose.

It is likely that while the code intended that checksums are optional (though required by default), it may in fact fail if some checksum files are not downloadable. With the current goal of indeed required availability of these files, behavior of the script otherwise was not much investigated and if there are issues in this corner - they are low-priority now.

**Comment by jimklimov** [comment]
A few issues were found while testing; now the script seems to work reasonably on IPCs and OVAs (and to some extent containers - it can download and check files, but this does not update the container itself).

One notable flaw is (and was previously) the direct unconditional use of `etn-sw-update` that is only in value-added OS images, and not a conditional use of upgradability validation wrapper if present in the new candidate image. Oh, and also that the tool used is from this current old OS image, not from the new candidate (ensuring *its* new policies about the meaning of valid upgrade and proper cleanup are followed) as was the original plan. This got backlogged now, and practical issues that were seen to follow from this discrepancy were plugged in this PR.

**Pull request merged**